thorlcr/master/thgraphmanager.cpp

```diff
@@ -1215,35 +1218,42 @@ void closeThorServerStatus()
  *  0 = unrecognised format, or wuid mismatch
  *  1 = success. new graph/wuid received.
  */
-static int recvNextGraph(unsigned timeoutMs, const char *wuid, StringBuffer &retWuid, StringBuffer &retGraphName)
+static int recvNextGraph(IJobQueue *thorQueue, unsigned timeoutMs, const char *wuid, StringBuffer &retWuid, StringBuffer &retGraphName)
 {
-    PROGLOG("Lingering time left: %.2f", ((float)timeoutMs)/1000);
-    CMessageBuffer msg;
-    if (!queryWorldCommunicator().recv(msg, NULL, MPTAG_THOR, nullptr, timeoutMs))
-        return -1;
     StringBuffer next;
-    msg.read(next);
+    CMessageBuffer msg;
+    if (thorQueue)
+    {
+        Owned<IJobQueueItem> item = thorQueue->dequeue(timeoutMs);
```

Comment:
question: If thor is closed down does this cleanly exit?

Reply:
good question, nothing explicitly cancel/aborts it, and probably needs to.
I'll verify.

Reply:
Have changed so queue will be cancelled on an abort condition.